src/file.ts

```diff
@@ -26,14 +27,71 @@ import fs from 'fs';
 import path from 'path';
 import { SONAR_CACHE_DIR } from './constants';
 import { log, LogLevel } from './logging';
+import { ScannerProperties, ScannerProperty } from './types';
+import { getHttpAgents } from './http-agent';
+import { getProxyUrl } from './proxy';
+import { fetch } from './request';
+import { promisify } from 'util';
+import * as stream from 'stream';
+
+const finished = promisify(stream.finished);
+
+export async function downloadFile(
```

Comment:
(to answer your comment in my PR)

To me it would make sense to have
- a function that only downloads a file
- a function which downloads a file + caches it + checks the checksum

wdyt? whoever merging second could make the change?